diff --git a/configure b/configure
index 9552a5c..c36e6f1 100755
--- a/configure
+++ b/configure
@@ -1492,6 +1492,8 @@ Optional Features:
   --enable-fast-install[=PKGS]
                           optimize for fast installation [default=yes]
   --disable-libtool-lock  avoid locking (might break parallel builds)
+  --enable-unittest=yes|no
+                          enable the unittest. requires boost (default is yes)
 
 Optional Packages:
   --with-PACKAGE[=ARG]    use PACKAGE [ARG=yes]
@@ -2447,7 +2449,6 @@ fi
 
 
 
-
 EXEMPI_REVISION=0
 
 EXEMPI_AGE=1
@@ -4839,7 +4840,7 @@ ia64-*-hpux*)
   ;;
 *-*-irix6*)
   # Find out which ABI we are using.
-  echo '#line 4842 "configure"' > conftest.$ac_ext
+  echo '#line 4843 "configure"' > conftest.$ac_ext
   if { (eval echo "$as_me:$LINENO: \"$ac_compile\"") >&5
   (eval $ac_compile) 2>&5
   ac_status=$?
@@ -7104,11 +7105,11 @@ else
    -e 's:.*FLAGS}\{0,1\} :&$lt_compiler_flag :; t' \
    -e 's: [^ ]*conftest\.: $lt_compiler_flag&:; t' \
    -e 's:$: $lt_compiler_flag:'`
-   (eval echo "\"\$as_me:7107: $lt_compile\"" >&5)
+   (eval echo "\"\$as_me:7108: $lt_compile\"" >&5)
    (eval "$lt_compile" 2>conftest.err)
    ac_status=$?
    cat conftest.err >&5
-   echo "$as_me:7111: \$? = $ac_status" >&5
+   echo "$as_me:7112: \$? = $ac_status" >&5
    if (exit $ac_status) && test -s "$ac_outfile"; then
      # The compiler can only warn and ignore the option if not recognized
      # So say no if there are warnings other than the usual output.
@@ -7394,11 +7395,11 @@ else
    -e 's:.*FLAGS}\{0,1\} :&$lt_compiler_flag :; t' \
    -e 's: [^ ]*conftest\.: $lt_compiler_flag&:; t' \
    -e 's:$: $lt_compiler_flag:'`
-   (eval echo "\"\$as_me:7397: $lt_compile\"" >&5)
+   (eval echo "\"\$as_me:7398: $lt_compile\"" >&5)
    (eval "$lt_compile" 2>conftest.err)
    ac_status=$?
    cat conftest.err >&5
-   echo "$as_me:7401: \$? = $ac_status" >&5
+   echo "$as_me:7402: \$? = $ac_status" >&5
    if (exit $ac_status) && test -s "$ac_outfile"; then
      # The compiler can only warn and ignore the option if not recognized
      # So say no if there are warnings other than the usual output.
@@ -7498,11 +7499,11 @@ else
    -e 's:.*FLAGS}\{0,1\} :&$lt_compiler_flag :; t' \
    -e 's: [^ ]*conftest\.: $lt_compiler_flag&:; t' \
    -e 's:$: $lt_compiler_flag:'`
-   (eval echo "\"\$as_me:7501: $lt_compile\"" >&5)
+   (eval echo "\"\$as_me:7502: $lt_compile\"" >&5)
    (eval "$lt_compile" 2>out/conftest.err)
    ac_status=$?
    cat out/conftest.err >&5
-   echo "$as_me:7505: \$? = $ac_status" >&5
+   echo "$as_me:7506: \$? = $ac_status" >&5
    if (exit $ac_status) && test -s out/conftest2.$ac_objext
    then
      # The compiler can only warn and ignore the option if not recognized
@@ -9847,7 +9848,7 @@ else
   lt_dlunknown=0; lt_dlno_uscore=1; lt_dlneed_uscore=2
   lt_status=$lt_dlunknown
   cat > conftest.$ac_ext <<EOF
-#line 9850 "configure"
+#line 9851 "configure"
 #include "confdefs.h"
 
 #if HAVE_DLFCN_H
@@ -9947,7 +9948,7 @@ else
   lt_dlunknown=0; lt_dlno_uscore=1; lt_dlneed_uscore=2
   lt_status=$lt_dlunknown
   cat > conftest.$ac_ext <<EOF
-#line 9950 "configure"
+#line 9951 "configure"
 #include "confdefs.h"
 
 #if HAVE_DLFCN_H
@@ -12367,11 +12368,11 @@ else
    -e 's:.*FLAGS}\{0,1\} :&$lt_compiler_flag :; t' \
    -e 's: [^ ]*conftest\.: $lt_compiler_flag&:; t' \
    -e 's:$: $lt_compiler_flag:'`
-   (eval echo "\"\$as_me:12370: $lt_compile\"" >&5)
+   (eval echo "\"\$as_me:12371: $lt_compile\"" >&5)
    (eval "$lt_compile" 2>conftest.err)
    ac_status=$?
    cat conftest.err >&5
-   echo "$as_me:12374: \$? = $ac_status" >&5
+   echo "$as_me:12375: \$? = $ac_status" >&5
    if (exit $ac_status) && test -s "$ac_outfile"; then
      # The compiler can only warn and ignore the option if not recognized
      # So say no if there are warnings other than the usual output.
@@ -12471,11 +12472,11 @@ else
    -e 's:.*FLAGS}\{0,1\} :&$lt_compiler_flag :; t' \
    -e 's: [^ ]*conftest\.: $lt_compiler_flag&:; t' \
    -e 's:$: $lt_compiler_flag:'`
-   (eval echo "\"\$as_me:12474: $lt_compile\"" >&5)
+   (eval echo "\"\$as_me:12475: $lt_compile\"" >&5)
    (eval "$lt_compile" 2>out/conftest.err)
    ac_status=$?
    cat out/conftest.err >&5
-   echo "$as_me:12478: \$? = $ac_status" >&5
+   echo "$as_me:12479: \$? = $ac_status" >&5
    if (exit $ac_status) && test -s out/conftest2.$ac_objext
    then
      # The compiler can only warn and ignore the option if not recognized
@@ -14033,11 +14034,11 @@ else
    -e 's:.*FLAGS}\{0,1\} :&$lt_compiler_flag :; t' \
    -e 's: [^ ]*conftest\.: $lt_compiler_flag&:; t' \
    -e 's:$: $lt_compiler_flag:'`
-   (eval echo "\"\$as_me:14036: $lt_compile\"" >&5)
+   (eval echo "\"\$as_me:14037: $lt_compile\"" >&5)
    (eval "$lt_compile" 2>conftest.err)
    ac_status=$?
    cat conftest.err >&5
-   echo "$as_me:14040: \$? = $ac_status" >&5
+   echo "$as_me:14041: \$? = $ac_status" >&5
    if (exit $ac_status) && test -s "$ac_outfile"; then
      # The compiler can only warn and ignore the option if not recognized
      # So say no if there are warnings other than the usual output.
@@ -14137,11 +14138,11 @@ else
    -e 's:.*FLAGS}\{0,1\} :&$lt_compiler_flag :; t' \
    -e 's: [^ ]*conftest\.: $lt_compiler_flag&:; t' \
    -e 's:$: $lt_compiler_flag:'`
-   (eval echo "\"\$as_me:14140: $lt_compile\"" >&5)
+   (eval echo "\"\$as_me:14141: $lt_compile\"" >&5)
    (eval "$lt_compile" 2>out/conftest.err)
    ac_status=$?
    cat out/conftest.err >&5
-   echo "$as_me:14144: \$? = $ac_status" >&5
+   echo "$as_me:14145: \$? = $ac_status" >&5
    if (exit $ac_status) && test -s out/conftest2.$ac_objext
    then
      # The compiler can only warn and ignore the option if not recognized
@@ -16324,11 +16325,11 @@ else
    -e 's:.*FLAGS}\{0,1\} :&$lt_compiler_flag :; t' \
    -e 's: [^ ]*conftest\.: $lt_compiler_flag&:; t' \
    -e 's:$: $lt_compiler_flag:'`
-   (eval echo "\"\$as_me:16327: $lt_compile\"" >&5)
+   (eval echo "\"\$as_me:16328: $lt_compile\"" >&5)
    (eval "$lt_compile" 2>conftest.err)
    ac_status=$?
    cat conftest.err >&5
-   echo "$as_me:16331: \$? = $ac_status" >&5
+   echo "$as_me:16332: \$? = $ac_status" >&5
    if (exit $ac_status) && test -s "$ac_outfile"; then
      # The compiler can only warn and ignore the option if not recognized
      # So say no if there are warnings other than the usual output.
@@ -16614,11 +16615,11 @@ else
    -e 's:.*FLAGS}\{0,1\} :&$lt_compiler_flag :; t' \
    -e 's: [^ ]*conftest\.: $lt_compiler_flag&:; t' \
    -e 's:$: $lt_compiler_flag:'`
-   (eval echo "\"\$as_me:16617: $lt_compile\"" >&5)
+   (eval echo "\"\$as_me:16618: $lt_compile\"" >&5)
    (eval "$lt_compile" 2>conftest.err)
    ac_status=$?
    cat conftest.err >&5
-   echo "$as_me:16621: \$? = $ac_status" >&5
+   echo "$as_me:16622: \$? = $ac_status" >&5
    if (exit $ac_status) && test -s "$ac_outfile"; then
      # The compiler can only warn and ignore the option if not recognized
      # So say no if there are warnings other than the usual output.
@@ -16718,11 +16719,11 @@ else
    -e 's:.*FLAGS}\{0,1\} :&$lt_compiler_flag :; t' \
    -e 's: [^ ]*conftest\.: $lt_compiler_flag&:; t' \
    -e 's:$: $lt_compiler_flag:'`
-   (eval echo "\"\$as_me:16721: $lt_compile\"" >&5)
+   (eval echo "\"\$as_me:16722: $lt_compile\"" >&5)
    (eval "$lt_compile" 2>out/conftest.err)
    ac_status=$?
    cat out/conftest.err >&5
-   echo "$as_me:16725: \$? = $ac_status" >&5
+   echo "$as_me:16726: \$? = $ac_status" >&5
    if (exit $ac_status) && test -s out/conftest2.$ac_objext
    then
      # The compiler can only warn and ignore the option if not recognized
@@ -19666,7 +19667,15 @@ fi
 
 
 
-     boost_version_major=1
+# Check whether --enable-unittest was given.
+if test "${enable_unittest+set}" = set; then
+  enableval=$enable_unittest; ENABLE_UNITTEST=$enableval
+else
+  ENABLE_UNITTEST=yes
+fi
+
+if test x$ENABLE_UNITTEST = xyes ; then
+        boost_version_major=1
      boost_version_minor=33
      boost_version_subminor=0
 
@@ -19801,7 +19810,7 @@ fi
 { echo "$as_me:$LINENO: result: $boost_cv_lib_version" >&5
 echo "${ECHO_T}$boost_cv_lib_version" >&6; }
 
-{ echo "$as_me:$LINENO: checking for the toolset name used by Boost for $CXX" >&5
+   { echo "$as_me:$LINENO: checking for the toolset name used by Boost for $CXX" >&5
 echo $ECHO_N "checking for the toolset name used by Boost for $CXX... $ECHO_C" >&6; }
 if test "${boost_cv_lib_tag+set}" = set; then
   echo $ECHO_N "(cached) $ECHO_C" >&6
@@ -20316,6 +20325,10 @@ ac_link='$CC -o conftest$ac_exeext $CFLAGS $CPPFLAGS $LDFLAGS conftest.$ac_ext $
 ac_compiler_gnu=$ac_cv_c_compiler_gnu
 
 
+else
+    ENABLE_UNINITEST=no
+fi
+
 
 case $build_vendor in
 	apple)
@@ -20330,7 +20343,7 @@ esac
 
 
 
- if test "x$BOOST_CPPFLAGS" != "x"; then
+ if test x$ENABLE_UNITTEST = xyes; then
   WITH_UNIT_TEST_TRUE=
   WITH_UNIT_TEST_FALSE='#'
 else
@@ -21870,4 +21883,5 @@ Defines:
    EXEMPI_PLATFORM_DEF = $EXEMPI_PLATFORM_DEF
    CPPFLAGS            = $CPPFLAGS
    LDFLAGS             = $LDFLAGS
+   Run unit test       = $ENABLE_UNITTEST
 "
diff --git a/configure.ac b/configure.ac
index f86d049..6425211 100644
--- a/configure.ac
+++ b/configure.ac
@@ -3,7 +3,6 @@ AC_INIT(exempi/xmp.h)
 AM_INIT_AUTOMAKE(exempi, 1.99.7)
 AM_MAINTAINER_MODE
 
-
 dnl ---------------------------------------------------------------------------
 dnl Advanced information about versioning:
 dnl   * "Writing shared libraries" by Mike Hearn
@@ -84,8 +83,18 @@ AC_CHECK_HEADER(expat.h, ,
 AC_CHECK_HEADER(iconv.h, ,
 	 AC_MSG_ERROR([iconv headers missing]))
 
-BOOST_REQUIRE([1.33.0])
-BOOST_TEST([mt-s])
+AC_ARG_ENABLE(unittest,
+              AC_HELP_STRING([--enable-unittest=yes|no],
+                            [enable the unittest. requires boost (default is yes)]),
+              ENABLE_UNITTEST=$enableval,
+              ENABLE_UNITTEST=yes)
+if test x$ENABLE_UNITTEST = xyes ; then
+   BOOST_REQUIRE([1.33.0])
+   BOOST_TEST([mt-s])
+else
+    ENABLE_UNINITEST=no
+fi
+
 
 case $build_vendor in
 	apple)
@@ -100,7 +109,7 @@ esac
 
 AC_SUBST(EXEMPI_PLATFORM_DEF)
 
-AM_CONDITIONAL(WITH_UNIT_TEST, test "x$BOOST_CPPFLAGS" != "x")
+AM_CONDITIONAL(WITH_UNIT_TEST, test x$ENABLE_UNITTEST = xyes)
 
 dnl Check for valgrind (optional) for make check.
 AC_CHECK_PROG(VALGRIND, valgrind, valgrind)
@@ -150,4 +159,5 @@ Defines:
    EXEMPI_PLATFORM_DEF = $EXEMPI_PLATFORM_DEF
    CPPFLAGS            = $CPPFLAGS
    LDFLAGS             = $LDFLAGS
+   Run unit test       = $ENABLE_UNITTEST
 "
